lib/ModelMixins/CatalogFunctionJobMixin.ts

```diff
@@ -130,6 +130,7 @@ function CatalogFunctionJobMixin<
      */
     protected abstract async _invoke(): Promise<boolean>;
 
+    @action
```

Comment:
Yeah possibly! I haven't used it before, but happy to look at it.

Reply:
There is an example of it here
https://github.com/TerriaJS/terriajs/blob/6702000db4b9908d1db2e23ef4b7c28a550ed984/lib/ReactViews/DragDropFile.tsx#L29-L66

But it may be easier to get rid of `@action` just wrap all synchronous calls inside the function with `runInAction`

Reply:
@nf-s I have wrapped the `_invoke()` call in an action. Let me know if that works. I think I had it in a working branch before but somehow missed it when splitting into separate PRs.

Reply:
Awesome thanks @na9da !